From 9e9379eebd136243a163e6075b427f6408e4c0e3 Mon Sep 17 00:00:00 2001
From: Sivashankari Madhavan <quic_sivamadh@quicinc.com>
Date: Thu, 1 Dec 2022 07:11:00 +0530
Subject: [PATCH] ath12k: Avoid multiple beacon countdown update

During the Bss color change, unwantedly updating the beacon countdown two
times for every separate beacon template in 6G. Due to this, some of the time it
reaches the 0 countdown and throws the WARNON.

[ 8393.129435] lr : ieee80211_beacon_update_cntdwn+0x5c/0x7c [mac80211]

Fix it by avoiding the multiple beacon coundown update.

Signed-off-by: Sivashankari Madhavan <quic_sivamadh@quicinc.com>
---
 drivers/net/wireless/ath/ath12k/mac.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/net/wireless/ath/ath12k/mac.c b/drivers/net/wireless/ath/ath12k/mac.c
index fe4be83..43b63d7 100644
--- a/drivers/net/wireless/ath/ath12k/mac.c
+++ b/drivers/net/wireless/ath/ath12k/mac.c
@@ -1506,7 +1506,7 @@ void ath12k_mac_bcn_tx_event(struct ath12k_vif *arvif)
 
 	arvif->bcca_zero_sent = false;
 
-	if (vif->color_change_active)
+	if (vif->color_change_active && !vif->bss_conf.ema_ap)
 		ieee80211_beacon_update_cntdwn(vif);
 	ieee80211_queue_work(ar->ah->hw, &arvif->update_bcn_template_work);
 }
-- 
2.17.1

